src/calculus.jl

```diff
@@ -0,0 +1,10 @@
+
+
+function diff(b1::Basic, b2::Basic)
```

Comment:
Is it possible to have a type hierarchy in Basic, so that we can use `diff(b1::Basic, b2::Symbol)`, where Symbol is a sub class of Basic?

What we do in Python is that, we get the object and then depending on the runtime type, we create a Symbol or an Integer or an Add or a Mul or a Pow, etc which contains the object.

How can we do this Julia?